src/helpers.ts

```diff
@@ -1,21 +1,27 @@
 export const getQueryParamsAsObject = (search: string) => {
-    const params: Record<string, string> = {}
+    const params: Record<string, string|string[]> = {}
 
-    new URLSearchParams(search).forEach((value, key) => params[key] = value)
+    new URLSearchParams(search).forEach((_, key) => {
+        const keyValues = new URLSearchParams(search).getAll(key)
+        const keyName = key.includes('[]') ? key.replace('[]','') : key
+        params[keyName] = keyName !== key ? keyValues.filter((value) => value !== '' || keyValues.length > 1) : keyValues[0]
+    })
 
     return params
 }
```

Comment:
I expected all the changes to be just here, in `getQueryParamsAsObject` and `objectToQueryParams`, and of course in the types.

I don't see a reason to create another `useArrayQueryStringKey`. We want to just change the implementation **inside** the core code, and keep the library interface as is.

Please explain if you think it impossible.

Reply:
Hi @baruchiro, I didn't create another hook to work with array query params.
I implemented the solution in the same hook "useQueryStringKey". The only difference is that instead of setting the value as a string, we can set an array of strings too.